rlp/main.cpp

```diff
@@ -256,14 +256,22 @@ int main(int argc, char** argv)
         prefs.escapeAll = true;
 
     bytes in;
-    if (inputFile == "--")
-        for (int i = cin.get(); i != -1; i = cin.get())
-            in.push_back((byte)i);
-    else if (boost::filesystem::is_regular_file(inputFile))
-        in = contents(inputFile);
-    else
-        in = asBytes(inputFile);
-
+    try
+    {
+        if (inputFile == "--")
+            for (int i = cin.get(); i != -1; i = cin.get())
+                in.push_back((byte)i);
+        else if (boost::filesystem::is_regular_file(inputFile))
+            in = contents(inputFile);
+    }
+    catch(const boost::filesystem::filesystem_error& _e)
+    {
+        //filesyatem_error contains lot of error flags but I couldn't find file name too long
+        //the closest I got was boost::system::errc -> enum errc_t ->  filename_too_long = ENAMETOOLONG
+        if(_e.what() == "File name too long") //don't know if this will work. Let's say temporary workaround.
```

Comment:
Because `what()` returns a `char *`  you can't compare directly to a string literal here.  You need to wrap the result of `what` in a string object before doing the comparison.

Reply:
Thanks for pointing out. My bad, sorry.
But the problem still remains unaddressed, if what I am doing (Directly comparing strings) will solve this issue.
I will correct my error soon.